<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PhysicsMainWindow</class>
 <widget class="QMainWindow" name="PhysicsMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1075</width>
    <height>697</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="maximumSize">
       <size>
        <width>350</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Contols</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <property name="spacing">
        <number>2</number>
       </property>
       <item row="3" column="0">
        <widget class="CopterInputsWidget" name="inputsWidget" native="true"/>
       </item>
       <item row="0" column="0">
        <widget class="QComboBox" name="comboBox">
         <property name="currentText">
          <string/>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QComboBox" name="comboBox_2">
         <property name="currentText">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="updateCameraButton">
         <property name="text">
          <string>Update cameras</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>3D View</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="CloudViewDialog" name="cloud" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Camera</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="AdvancedImageWidget" name="imageView" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1075</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu12">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionShowLog"/>
   </widget>
   <widget class="QMenu" name="menuCamera">
    <property name="title">
     <string>Ca&amp;mera</string>
    </property>
    <addaction name="actionCameraInput"/>
    <addaction name="actionCameraStart"/>
    <addaction name="actionCameraPause"/>
    <addaction name="actionCameraStop"/>
    <addaction name="actionCameraCaptureParamteres"/>
    <addaction name="actionCalibrate"/>
    <addaction name="actionLoad_Calibration"/>
   </widget>
   <widget class="QMenu" name="menuJoystick">
    <property name="title">
     <string>&amp;Joystick</string>
    </property>
    <addaction name="actionJoystickParameters"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuProcessing">
    <property name="title">
     <string>Processi&amp;ng</string>
    </property>
    <addaction name="actionCameraModel"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionNewBackend"/>
   </widget>
   <addaction name="menu12"/>
   <addaction name="menuView"/>
   <addaction name="menuCamera"/>
   <addaction name="menuJoystick"/>
   <addaction name="menuProcessing"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetMovable|QDockWidget::DockWidgetVerticalTitleBar</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QToolButton" name="toolButton_3">
       <property name="text">
        <string>connect to real device</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/new/prefix1/connect.png</normaloff>:/new/prefix1/connect.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextUnderIcon</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="connetToVirtualButton">
       <property name="text">
        <string>connect to virtual</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/new/prefix1/server_connect.png</normaloff>:/new/prefix1/server_connect.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextUnderIcon</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="toolButton_2">
       <property name="text">
        <string>AUTOPILOT
(press any js key to stop)</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/new/prefix1/robot.png</normaloff>:/new/prefix1/robot.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextUnderIcon</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>imTest</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/new/colors/colors/color_blue.png</normaloff>:/new/colors/colors/color_blue.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="JoyButton">
       <property name="text">
        <string>JoyStick mode</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/new/prefix1/controller.png</normaloff>:/new/prefix1/controller.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextUnderIcon</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Output from JS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="iiOutputSlider">
       <property name="minimum">
        <number>-1</number>
       </property>
       <property name="maximum">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Output from II</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="logDockWidget">
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="LoggerWidget" name="logWidget"/>
  </widget>
  <action name="actionAbout">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/info_rhombus.png</normaloff>:/new/prefix1/info_rhombus.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionCameraInput">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/webcam_add.png</normaloff>:/new/prefix1/webcam_add.png</iconset>
   </property>
   <property name="text">
    <string>Camera Input</string>
   </property>
  </action>
  <action name="actionCameraStart">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/play.png</normaloff>:/new/prefix1/play.png</iconset>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
  </action>
  <action name="actionCameraPause">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/pause.png</normaloff>:/new/prefix1/pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
  </action>
  <action name="actionCameraStop">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/stop_2.png</normaloff>:/new/prefix1/stop_2.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionJoystickParameters">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/joystick_error.png</normaloff>:/new/prefix1/joystick_error.png</iconset>
   </property>
   <property name="text">
    <string>Parameters</string>
   </property>
  </action>
  <action name="actionCameraModel">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/webcam_add.png</normaloff>:/new/prefix1/webcam_add.png</iconset>
   </property>
   <property name="text">
    <string>Camera Model</string>
   </property>
  </action>
  <action name="actionNewBackend">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/flag_start.png</normaloff>:/new/prefix1/flag_start.png</iconset>
   </property>
   <property name="text">
    <string>New Backend</string>
   </property>
   <property name="toolTip">
    <string>New Backend</string>
   </property>
  </action>
  <action name="actionShowLog">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/application_view_list.png</normaloff>:/new/prefix1/application_view_list.png</iconset>
   </property>
   <property name="text">
    <string>Show Log</string>
   </property>
  </action>
  <action name="actionCameraCaptureParamteres">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/cctv_camera.png</normaloff>:/new/prefix1/cctv_camera.png</iconset>
   </property>
   <property name="text">
    <string>Camera Capture Paramteres</string>
   </property>
  </action>
  <action name="actionCalibrate">
   <property name="text">
    <string>Calibrate</string>
   </property>
  </action>
  <action name="actionLoad_Calibration">
   <property name="text">
    <string>Load Calibration</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CloudViewDialog</class>
   <extends>QWidget</extends>
   <header location="global">cloudViewDialog.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AdvancedImageWidget</class>
   <extends>QWidget</extends>
   <header location="global">advancedImageWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CopterInputsWidget</class>
   <extends>QWidget</extends>
   <header>copterInputsWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LoggerWidget</class>
   <extends>QWidget</extends>
   <header>loggerWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionShowLog</sender>
   <signal>toggled(bool)</signal>
   <receiver>logDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>644</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCameraStart</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>startCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCameraStop</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>stopCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCameraPause</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>pauseCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCameraInput</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>showCameraInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCameraCaptureParamteres</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>showCameraParametersWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>showAboutWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionJoystickParameters</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>showJoystickSettingsWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCalibrate</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>CalibrateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_Calibration</sender>
   <signal>triggered()</signal>
   <receiver>PhysicsMainWindow</receiver>
   <slot>LoadCalibrationSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showCameraInput()</slot>
  <slot>startCamera()</slot>
  <slot>pauseCamera()</slot>
  <slot>stopCamera()</slot>
  <slot>showCameraParametersWidget()</slot>
  <slot>showCameraModelWidget()</slot>
  <slot>showGraphDialog()</slot>
  <slot>showAboutWidget()</slot>
  <slot>showJoystickSettingsWidget()</slot>
  <slot>CalibrateCamera()</slot>
  <slot>LoadCalibrationSettings()</slot>
 </slots>
</ui>
